zarr/storage.py

```diff
@@ -1378,13 +1378,16 @@ def __getitem__(self, key):
     def setitems(self, values):
         if self.mode == 'r':
             raise ReadOnlyError()
-        values = {self._normalize_key(key): val for key, val in values.items()}
+
+        # Normalize keys and make sure the values are bytes
+        values = {self._normalize_key(key): ensure_bytes(val) for key, val in values.items()}
         self.map.setitems(values)
```

Comment:
Maybe something like this would work

```suggestion
        norm_values = {}
        for key, val in values.items():
            norm_key = self._normalize_key(key)
            try:
                val = ensure_contiguous_ndarray_like(val)
            except TypeError:
                val = ensure_bytes(val)
            norm_values[norm_key] = val
        self.map.setitems(norm_values)
```